### PR TITLE
Update Dynu code and .ini file

### PR DESCRIPTION
**Pull request by panosangel**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/docker-swag/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:

Running SWAG for the first time I got errors while setting it up to work with Dynu DNS challenge. The code for `DNSPLUGIN` flag was wrongly named and the .ini config file had the wrong name.

## Benefits of this PR and context:

If it's not fixed, every single Dynu DNS user will have to make the same adjustments.

## How Has This Been Tested?

I kept making adjustments to `docker-compose.yml` and the filesystem until there were no errors shown.

### CODE REVIEW

**Comment by aptalca** [comment]
dynu is a different plugin and it's abandoned

**Comment by panosangel** [comment]
According to docker logs `DNSPLUGIN=dynudns` is not a recognized value and `dynu` is suggested. Could it be a validation issue?

```
swag  | [migrations] started
swag  | [migrations] 01-nginx-site-confs-default: executing...
swag  | [migrations] 01-nginx-site-confs-default: succeeded
swag  | [migrations] done
swag  | ───────────────────────────────────────
swag  | 
swag  |       ██╗     ███████╗██╗ ██████╗
swag  |       ██║     ██╔════╝██║██╔═══██╗
swag  |       ██║     ███████╗██║██║   ██║
swag  |       ██║     ╚════██║██║██║   ██║
swag  |       ███████╗███████║██║╚██████╔╝
swag  |       ╚══════╝╚══════╝╚═╝ ╚═════╝
swag  | 
swag  |    Brought to you by linuxserver.io
swag  | ───────────────────────────────────────
swag  | 
swag  | To support the app dev(s) visit:
swag  | Certbot: https://supporters.eff.org/donate/support-work-on-certbot
swag  | 
swag  | To support LSIO projects visit:
swag  | https://www.linuxserver.io/donate/
swag  | 
swag  | ───────────────────────────────────────
swag  | GID/UID
swag  | ───────────────────────────────────────
swag  | 
swag  | User UID:    1000
swag  | User GID:    1000
swag  | ───────────────────────────────────────
swag  | 
swag  | Setting resolver to  127.0.0.11
swag  | Setting worker_processes to 8
swag  | generating self-signed keys in /config/keys, you can replace these with your own keys if required
swag  | ....+...........+...+...+.......+..+.........+.+......+........+.+.....................+++++++++++++++++++++++++++++++++++++++*....+........+.............+..+.+..+...+....+.....+......+.+++++++++++++++++++++++++++++++++++++++*....+...++++++
swag  | ..+..+...+...+....+......+..+...+++++++++++++++++++++++++++++++++++++++*......+.......+.....+................+.....+.+......+...+++++++++++++++++++++++++++++++++++++++*.+..+...+......+.+...+...............+.....+......+...............................+..................+.....+.+......+.....+..........+...+............+...+.....+......+...++++++
swag  | -----
swag  | Variables set:
swag  | PUID=1000
swag  | PGID=1000
swag  | TZ=Europe/Athens
swag  | URL=domain.com
swag  | SUBDOMAINS=wildcard
swag  | EXTRA_DOMAINS=
swag  | ONLY_SUBDOMAINS=false
swag  | VALIDATION=dns
swag  | CERTPROVIDER=
swag  | DNSPLUGIN=dynudns
swag  | EMAIL=user@domain.com
swag  | STAGING=true
swag  | 
swag  | Please set the DNSPLUGIN variable to one of the following:
swag  | acmedns
swag  | aliyun
swag  | azure
swag  | bunny
swag  | cloudflare
swag  | cpanel
swag  | desec
swag  | digitalocean
swag  | directadmin
swag  | dnsimple
swag  | dnsmadeeasy
swag  | dnspod
swag  | do
swag  | domeneshop
swag  | dreamhost
swag  | duckdns
swag  | dynu
swag  | freedns
swag  | gandi
swag  | gehirn
swag  | glesys
swag  | godaddy
swag  | google
swag  | google-domains
swag  | he
swag  | hetzner
swag  | infomaniak
swag  | inwx
swag  | ionos
swag  | linode
swag  | loopia
swag  | luadns
swag  | namecheap
swag  | netcup
swag  | njalla
swag  | nsone
swag  | ovh
swag  | porkbun
swag  | rfc2136
swag  | route53
swag  | sakuracloud
swag  | standalone
swag  | transip
swag  | vultr
```

**Comment by lucas-cornette** [comment]
I'm also using dynudns and can confirm these changes have to be made to be able to use the dynu plugin.
I have renamed the ./config/dns-conf/dynu-credentials.ini file to dynu.ini and changed the DNSPLUGIN variable in my dockerfile from dynudns to dynu. After these changes, certbot is successfully generating certs for my domain.

File: README.md [hunk 1]
```diff
@@ -227,7 +227,7 @@ Containers are configured using parameters passed at runtime (such as those abov
...
+| `-e DNSPLUGIN=cloudflare` | Required if `VALIDATION` is set to `dns`. Options are `acmedns`, `aliyun`, `azure`, `bunny`, `cloudflare`, `cpanel`, `desec`, `digitalocean`, `directadmin`, `dnsimple`, `dnsmadeeasy`, `dnspod`, `do`, `domeneshop`, `dreamhost`, `duckdns`, `dynu`, `freedns`, `gandi`, `gehirn`, `glesys`, `godaddy`, `google`, `google-domains`, `he`, `hetzner`, `infomaniak`, `inwx`, `ionos`, `linode`, `loopia`, `luadns`, `namecheap`, `netcup`, `njalla`, `nsone`, `ovh`, `porkbun`, `rfc2136`, `route53`, `sakuracloud`, `standalone`, `transip`, and `vultr`. Also need to enter the credentials into the corresponding ini (or json for some plugins) file under `/config/dns-conf`. |
```
Comment (nemchik) [inline README.md:230]: This change needs to happen in the `readme-vars.yml`
https://github.com/panosangel/docker-swag/blob/05b586d6df5df4fb3d648db62de9273bd3329190/readme-vars.yml#L53

**Review by nemchik — CHANGES_REQUESTED** [review]
Noting that https://pypi.org/project/certbot-dns-dynudns/ and https://github.com/DustyRah/certbot-dns-dynudns/ do not mention the new config file name, but I'm guessing the old plugin that was deprecated in https://github.com/linuxserver/docker-swag/pull/439 was due to not being updated to newer code standards in certbot, and the new plugin introduced in https://github.com/linuxserver/docker-swag/pull/463 may have been a fork of the original that updated the code but maybe did not update the readme. I'm not positive, but the rename makes sense given how other current plugins work with the config naming.

**Comment by aptalca** [comment]
This change is not desired. There are two separate dynu plugins. One is old and abandoned (`dynu`), and the other is new and maintained by the dynu employees (`dynudns`).

The old and abandoned plugin force downgrades a bunch of the certbot dependencies because it has old deps hardcoded. We do not want that because it affects everyone, not just dynu users.

It seems the real issue here is, when the dynu employees forked the abandoned plugin and published it under a different name, they only changed the name externally (package name). Internally it still uses the old references.

This requires putting in an exception to use the new plugin package but with the old internal name. I'm short on time these days and have no ability to test and trouble shoot so it will have to come from someone else.

**Comment by nemchik** [comment]
> It seems the real issue here is, when the dynu employees forked the abandoned plugin and published it under a different name, they only changed the name externally (package name). Internally it still uses the old references.
> 
> This requires putting in an exception to use the new plugin package but with the old internal name. I'm short on time these days and have no ability to test and trouble shoot so it will have to come from someone else.
> 
> EDIT:
> 
> I take it back. While most of what I wrote above is true, I took another look at the full log and now see that it still has the new plugin installed, but it references the old plugin's name. I assumed it would force install the old plugin on startup but doesn't. So while not a proper solution, I guess it would work.

Yeah it looks like we removed the old plugin in https://github.com/linuxserver/docker-swag/pull/439 and added the new plugin in https://github.com/linuxserver/docker-swag/pull/463 but there was some confusion on our part.

https://github.com/linuxserver/docker-swag/pull/463/commits/4b4c103df40d6c5e223cd98cbd3447f310b9d5c7 added `root/defaults/dns-conf/dynu-credentials.ini` which should have been `root/defaults/dns-conf/dynudns.ini` because of how our init handles the credentials file here https://github.com/linuxserver/docker-swag/blob/db6fbc2731a1824e4cdd7d8c59a4e05603a528a9/root/etc/s6-overlay/s6-rc.d/init-certbot-config/run#L292 and here https://github.com/linuxserver/docker-swag/blob/db6fbc2731a1824e4cdd7d8c59a4e05603a528a9/root/etc/s6-overlay/s6-rc.d/init-certbot-config/run#L299C53-L299C70

Apparently even if we had done the above correctly, the new `dynudns` plugin's entrypoint is still `dns-dynu` as shown here https://github.com/DustyRah/certbot-dns-dynudns/blob/da9df1b91f4615c8e83a7213a701dd4b87e3505c/setup.py#L69 and our init uses that entrypoint for the `authenticator` in certbot's `cli.ini` here https://github.com/linuxserver/docker-swag/blob/db6fbc2731a1824e4cdd7d8c59a4e05603a528a9/root/etc/s6-overlay/s6-rc.d/init-certbot-config/run#L298 so despite the package name being `dynudns` the correct naming we need here is just `dynu`.

**Review by nemchik — CHANGES_REQUESTED** [review]
Merge conflicts need to be resolved.

The changes in `README.md` can be completely reverted (no need to fix the merge conflict in this file, if you just revert it entirely our build system will update `README.md` from `readme-vars.yml`).

`readme-vars.yml` needs a changelog entry.

**Comment by panosangel** [comment]
To be honest I'm kind of confused which approach we'll take now so I can make the appropriate adjustments.

**Comment by nemchik** [comment]
> To be honest I'm kind of confused which approach we'll take now so I can make the appropriate adjustments.

I pushed some adjustments for you to resolve the merge conflict and add a changelog entry.

**Review by nemchik — APPROVED** [review]

**Review by aptalca — APPROVED** [review]

**Pull request merged**